camel/environments/tic_tac_toe.py

```
@@ -254,11 +258,27 @@ async def _update_state(self, action: Action) -> None:
 
-        # Attempt to parse the agent's chosen move
-        extraction_result = await self.extractor.extract(action.llm_response)
+         # Attempt to parse the agent's chosen move
+        try:
+            # Add timeout control
+            extraction_result = await asyncio.wait_for(
+                self.extractor.extract(action.llm_response),
+                timeout=self._timeout  # Use the timeout from MultiStepEnv
+            )
+        except asyncio.TimeoutError:
+            logger.error(f"Move extraction timed out after {self._timeout}s")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)